Enable capability of using http2 cleartext (h2c) when communicating with coprocessor (#5300)
apollographql/router: docs/source/customizations/coprocessor.mdx

In this case, the `RouterService` only sends a coprocessor request whenever it receives a client request. The coprocessor request body includes _no_ data related to the client request (only "control" data, which is [covered below](#coprocessor-request-format)).

### Client configuration

> You can control the coprocessor client configuration as needed.  At the moment only experimental_http2 is supported.

Reviewer: ```suggestion
<HttpConnection type="coprocessor" />
```

This should work fine once https://github.com/apollographql/docs/pull/889 merges.

(Tested locally.)

> You can control the coprocessor client configuration as needed.  At the moment only experimental_http2 is supported.

Reviewer: ```suggestion
<HttpConnection type="coprocessor" />
```

Introduced in https://github.com/apollographql/docs/pull/889 and used in https://github.com/apollographql/router/pull/5536.

(See deploy preview: https://deploy-preview-5536--apollo-router-docs.netlify.app/configuration/traffic-shaping#http2)

> For example to enable h2c (http2 cleartext) communication:

Reviewer: ```suggestion
For example, to enable h2c communication with a coprocessor you can use this configuration:
```

Reviewer 2: Applied via 305a2b53ff8a1e918194f86ff0687235aa82b4a6.  Somehow all suggestions were duplicated.  I tried to accept the right ones. ;)

> For example to enable h2c (http2 cleartext) communication:

Reviewer: ```suggestion
For example, to enable h2c (http2 cleartext) communication with a coprocessor you can use this configuration::
```

```yaml title="router.yaml"
coprocessor:

>   url: http://127.0.0.1:8081 # When using an http (not https) url and http2only will result in connections that use h2c
>   client:

Reviewer: Suggest moving comment to the next line for readability:

```suggestion
  url: http://127.0.0.1:8081
  # Using an HTTP (not HTTPS) URL and experimental_http2: http2only results in connections that use h2c
  client:
```

>   url: http://127.0.0.1:8081 # When using an http (not https) url and http2only will result in connections that use h2c
>   client:

Reviewer: Suggest moving comment to the next line for readability:

```suggestion
  url: http://127.0.0.1:8081
  # Using an HTTP (not HTTPS) URL and experimental_http2: http2only results in connections that use h2c
  client:
```
